File: Boxuan/16_01_17/Sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        left, right = 0, x
        
        while left + 1 < right:
            mid = left + (right - left) / 2
            if mid * mid == x:
                return mid
                
            elif mid * mid < x:
                left = mid
            
            else:
                right = mid 
                
                
        ans = right if right * right == x else left
        
        return ans

File: Boxuan/16_01_16/Sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) {
        // write your code here
        long left = 0;
        if (x == 1)
            return 1;
        long right = x;
        
        while (left+1 < right) {
            long mid = left + (right-left)/2;
            if (x > mid*mid) {
                left = mid;
            } else if (x < mid*mid) {
                right = mid;
            } else {
                return mid;
            }
        }
        return left;
    }
};